Makefile

```diff
@@ -76,31 +96,36 @@ ifeq ($(NVCC),)
     $(info nvcc not found, skipping CUDA builds)
 else
     $(info nvcc found, including CUDA builds)
-    TARGETS += train_gpt2cu test_gpt2cu train_gpt2fp32cu test_gpt2fp32cu
+    TARGETS += train_gpt2cu test_gpt2cu train_gpt2fp32cu test_gpt2fp32cu profile_gpt2cu
 endif
 
 all: $(TARGETS)
 
 train_gpt2: train_gpt2.c
-	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS) $< $(LDLIBS) -o $@
+	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS) $< $(LDLIBS) $(OUTPUT_FILE)
```

Comment:
My personal opinion is that I think the OUTPUT_FILE variable obfuscates a little more than it helps. It is also not used for the *cu recipies.

Reply:
Yes, that's because the Cuda linker is different (it's runs like a 2-pass tool) and could not locate the switch to control the name of the output file. I assume there is one - let me know if you know what the syntax is for that.

Reply:
Yes, nvcc is doing a lot of work under the hood. With --verbose one can see it. It relies on the compilers and linkers already installed... Good thing that NVCC is OK with the -o flag then.

Reply:
Yes, that surprised me that the -o worked. I will try to find a better solution.

Reply:
I've got this fixed - will update shortly.